### DISCUSSION TITLE
Best way to update vectors in collection

### DISCUSSION THREAD

**Discussion by VitalyPetrov** [milvus-io/milvus#12235] [Q&A and General discussion] [answered]
Hi!

I m looking for the best way to update the vectors in given collection by `item_id`.
Surely I can do it by deleting this collection and creating it again but it cause a certain delay of the dependent service to work properly.
Could you please clarify the best way for data updating?

Thanks,
Vitaly

**Comment by yhmo** [comment, marked as answer]
Currently milvus has no upsert() interface, neither v1.0 nor v2.0. To update a vector, the workaround is delete() + insert().

**Reply by VitalyPetrov** [reply]
Ok, that's what I afraid.
However, it seems that painless update can be performed by using rename_collectiion method/endpoint, which is planned on [Milvus v2.2](https://milvus.io/docs/roadmap.md). The pipeline is as following:

create collection named `temp` similar to the target one (say `item_vectors`) and insert vectors in it -> drop target collection and rename temporary collection to `item_vectors`.

It should cause much lesser lag.

**Reply by yhmo** [reply]
Do you mean "collection alias"?
In v2.0 RC6 we add a feature, allow user to specify an alias for a collection, and change this alias from this collection to another quickly.
The typically user case: 
1. in day 1, user create a collection "day_1", give an alias "A" to this collection
2. the client can do all things by this alias: insert("A"), search("A"), all these operations are passed to the "day_1"
3. in day 2, user create another collection "day_2", and assign the alias "A" to "day_2"
4. the client continue use the alias to do all things: insert("A"), search("A"), actually, all these operations are passed to the "day_2"

**Comment by Maghamrohith** [comment]
Hi @yhmo ,
Can we update an existing collection instead of delete and insert in python? currently? or any functionality added for that?
thanks in advance

**Reply by yhmo** [reply]
Do you mean to change the schema of an existing collection?
